src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_kernel_emitter.cpp

```diff
@@ -0,0 +1,360 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "jit_kernel_emitter.hpp"
+
+using namespace Xbyak_aarch64;
+
+namespace ov {
+namespace intel_cpu {
+namespace aarch64 {
+
+using jit_generator = dnnl::impl::cpu::aarch64::jit_generator;
+using cpu_isa_t = dnnl::impl::cpu::aarch64::cpu_isa_t;
+using ExpressionPtr = ov::snippets::lowered::ExpressionPtr;
+
+inline static std::vector<XReg> transform_idxs_to_regs(const std::vector<size_t>& idxs) {
+    std::vector<XReg> regs;
+    regs.resize(idxs.size(), XReg(0));
+    std::transform(idxs.begin(), idxs.end(), regs.begin(), [](size_t idx){return XReg(idx);});
+    return regs;
+}
+
+inline static std::vector<size_t> transform_snippets_regs_to_idxs(const std::vector<snippets::Reg>& regs) {
+    std::vector<size_t> idxs(regs.size());
+    std::transform(regs.cbegin(), regs.cend(), idxs.begin(), [](const snippets::Reg& reg) { return reg.idx; });
+    return idxs;
+}
+
+inline static void push_XReg(dnnl::impl::cpu::aarch64::jit_generator *h, uint32_t xreg_idx) {
```

Comment:
Applied then removed. Because while apply https://github.com/openvinotoolkit/openvino/pull/23180#discussion_r1514128355, I found that the callee-saved registers have already been saved in h->preamble() in jit_kernel_emitter::emit_impl. And if we want to save some other gprs in the future, we can directly use something like h->stp. Thanks Alexandra!